pkg/portal/middleware/aad.go

```diff
@@ -141,7 +142,18 @@ func (a *aad) AAD(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		session, err := a.store.Get(r, SessionName)
 		if err != nil {
-			a.internalServerError(w, err)
+			cookieError := err.(securecookie.Error)
+			if cookieError != nil && cookieError.IsDecode() {
+				cookie := &http.Cookie{
+					Name:    SessionName,
+					Path:    "/",
+					Expires: time.Unix(0, 0),
+				}
+				http.SetCookie(w, cookie)
+				http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
+			} else {
+				a.internalServerError(w, err)
+			}
```

Comment:
maybe we could also get rid of the else with a `return` after line 153. The else is not necessary here

Reply:
I agree on the general practice of not introducing unnecessary else's, but I am not sure how we can get rid of the `else` in this instance. We need the `a.internalServerError()` to catch all errors that are not a `cookieError.IsDecode()` error, with the return preventing any of the rest of the function from running.

Reply:
```

			cookieError, ok := err.(securecookie.Error)
			if ok && cookieError != nil && cookieError.IsDecode() {
				cookie := &http.Cookie{
					Name:    SessionName,
					Path:    "/",
					Expires: time.Unix(0, 0),
				}
				http.SetCookie(w, cookie)
				http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
				return
			} 
			a.internalServerError(w, err)
			return
```

Reply:
this is not what duplication is. Also , it may have 2 returns but is has less nesting which reduces complexity.

Reply:
I 100% agree with the removal of the "else", take a look at my review if you wish.

Reply:
I prefer the first option in cases when the condition follows logically to the else and the number of lines is low, because the logic this simple to parse.

The example you are using is contrived though, because its not a conditional on the error output. It is on whether to _log an error_ **or** _perform an action_. Obviously that reads better in the second example.

The code in this PR is conceptually doing this and it reads easily this way:
```go
if error-is-a-particular-type {
    handleIt()
} else {
    handleItDifferently()
}
return // to prevent the rest of the code running
```
This is a well practised approach to handling this kind of logic flow and does not benefit from the early-return approach to reduce nesting.

Reply:
@darthhexx  I had a chance to take a look at the comments. Some good discussion there. From my experience, it's always a good practice to eliminate unnecessary `else` s. Moreover, it's also considered a linting issue. One question though. I immediately noticed a return statement, which was also pointed by @facchettos  as well. Why do we even need that return statement at the end, couldn't we just return from the if block to stop further execution and wouldn't the inside function return any way when it has executed a.internalServerError(w, err) . Unless, the last return statement stops the outside function from executing further, which it doesn't, I think we should be able to get rid of the last return statement, move it inside if and remove the else.